tests/data/test_image.py

```diff
@@ -17,6 +17,10 @@ def test_image_not_found(self):
         with self.assertRaises(FileNotFoundError):
             ScalarImage('nopath')
 
+    def test_wrong_path_value(self):
+        with self.assertRaises(TypeError):
+            ScalarImage('~&./@#"!?X7=+')
```

Comment:
Looks like Obélix swearing haha.

However I'm not sure this should raise a `TypeError`. The `type` is `str`, which is fine. Why not `ValueError`?

Reply:
Yes I got inspired for this one haha.

The caught one is a `RuntimeError`, do you prefer to have a `ValueError` or a `RuntimeError` then?

Reply:
I think propagating that `RuntimeError` is the wise choice.